addons/sms/models/ir_actions_server.py

```diff
@@ -44,6 +44,15 @@ def _compute_name(self):
             )
         super(ServerActions, self - to_update)._compute_name()
 
+    @api.depends('state')
+    def _compute_available_model_ids(self):
+        mail_based = self.filtered(lambda action: action.state == 'sms')
+        if mail_based:
+            mail_models = self.env['ir.model'].search([('is_mail_thread', '=', True), ('transient', '=', False)])
+            for action in mail_based:
+                action.available_model_ids = mail_models.ids
```

Comment:
A bit strange that it bypasses "_get_allowed_models" I would say.

Reply:
It was just to have something in base to be honest, that is not "all ids". As anyway only admins can modify automated rules / server actions it is not that important I think.